src/util/settingsUtils.js

```diff
@@ -59,3 +59,29 @@ export function settingEnabled(data, enabledKey, startTimeKey, endTimeKey) {
 	const endTime = readDateSetting(data, endTimeKey);
 	return enabled && isWithinRange(new Date(), startTime, endTime);
 }
+
+/**
+ * Convert a string to a 32 bit integer
+ * Inspiration: https://stackoverflow.com/a/8831937
+ *
+ * @param {string} source
+ */
+export function hashCode(source) {
+	let hash = 0;
+	// return 0 if empty or incorrect type
+	if (typeof source === 'undefined' || typeof source !== 'string' || source === '') {
+		return hash;
+	}
+	// shift the bits
+	for (let i = 0; i < source.length; i++) { // eslint-disable-line
+		// isolate character
+		const character = source.charCodeAt(i);
+		// original approach left shift
+		hash = ((hash << 5) - hash) + character; // eslint-disable-line no-bitwise
```

Comment:
Bit shifting is going to be a lot more efficient, so I think you can go ahead and disable the no-bitwise check for the whole file